laws/src/main/scala/cats/laws/discipline/MiniFloat.scala

```diff
@@ -0,0 +1,177 @@
+package cats
+package laws
+package discipline
+
+import cats.implicits.{catsSyntaxPartialOrder, toTraverseFilterOps}
+import cats.kernel.{BoundedSemilattice, CommutativeMonoid}
+
+/**
+ * Similar to `Float`, but with a much smaller domain. The exact range of [[MiniFloat]] may be tuned from time to time,
+ * so consumers of this type should avoid depending on its exact range.
+ *
+ * `MiniFloat` has overflow and floating-point error characteristics similar to `Float`, but these are exaggerated due
+ * to its small domain. It is only approximately commutative and associative under addition and multiplication, due to
+ * floating-point errors, overflows, and the behaviour of `NaN`.
+ *
+ * Note that unlike `Float`, `MiniFloat` does not support the representation of negative zero (`-0f`).
+ */
+sealed abstract class MiniFloat private (val toFloat: Float) {
+  def toDouble: Double = toFloat.toDouble
+  def toInt: Int = toFloat.toInt
+  def toLong: Long = toFloat.toLong
+
+  def +(that: MiniFloat): MiniFloat = MiniFloat.from(this.toFloat + that.toFloat)
+  def -(that: MiniFloat): MiniFloat = MiniFloat.from(this.toFloat - that.toFloat)
+  def *(that: MiniFloat): MiniFloat = MiniFloat.from(this.toFloat * that.toFloat)
+  def /(that: MiniFloat): MiniFloat = MiniFloat.from(this.toFloat / that.toFloat)
+  def unary_- : MiniFloat = MiniFloat.from(-this.toFloat)
+
+  def isNaN: Boolean = toFloat.isNaN
+  def isFinite: Boolean = java.lang.Float.isFinite(toFloat)
+
+  override def toString = s"MiniFloat($toFloat)"
+
+  override def equals(other: Any): Boolean = other match {
+    case that: MiniFloat => this.toFloat == that.toFloat
+    case _               => false
+  }
+
+  override def hashCode: Int = java.lang.Float.hashCode(toFloat)
+
+}
+
+object MiniFloat {
+
+  object PositiveInfinity extends MiniFloat(Float.PositiveInfinity)
+  object NegativeInfinity extends MiniFloat(Float.NegativeInfinity)
+  object NaN extends MiniFloat(Float.NaN)
+
+  final private class Finite private (significand: Int, exponent: Int)
+      extends MiniFloat(significand * math.pow(Finite.base.toDouble, exponent.toDouble).toFloat)
+
+  private[MiniFloat] object Finite {
+
+    private[MiniFloat] val base = 2
+
+    private val minSignificand = -2
+    private val maxSignificand = 2
+
+    private val minExponent = -1
+    private val maxExponent = 2
```

Comment:
These define the range of `MiniFloat`. The finite values currently are:

  *  `-8.0`
  *  `-4.0`
  *  `-2.0`
  *  `-1.0`
  *  `-0.5`
  *  `0.0`
  *  `0.5`
  *  `1.0`
  *  `2.0`
  *  `4.0`
  *  `8.0`